docs/topics/auth/customizing.txt

```diff
@@ -444,6 +444,32 @@ a :class:`~django.db.models.ForeignKey` or
 :class:`~django.db.models.OneToOneField` to ``settings.AUTH_USER_MODEL``
 as described below.
 
+Customizing Username Validators
+-------------------------------
+
+.. currentmodule:: django.contrib.auth
+
+Django username validator can be customize using the ``settings.AUTH_USERNAME_VALIDATORS``. You can create validators.py and write your own username validator class. For example::
+
+
+    from django.core.validators import RegexValidator
+    
+    class CustomUsernameValidator(RegexValidator):
+        regex = r'^[\w.@+-']+\Z'
+        message = ("This value may contain only Enlight letters, numbers, and @/./+/-/'/_ charecters.")
+
+    .. note::
+
+        Username validator must subclass from :class:`~django.core.validators.RegexValidator` class.
+            
+
+Then, in ``settings.py`` file you can add ``AUTH_USERNAME_VALIDATORS`` and intantiate an object of CustomUsernameValidator. For example::
+
+    
+    from myapp.validators import CustomUsernameValidator                
+    
+    AUTH_USERNAME_VALIDATORS = [ CustomUsernameValidator() ]
```

Comment:
Here is where we shouldn't be asking users to instantiate objects in global_settings.py. Notice no other objects are instantiated there.